test/QuarkBuilderCometWithdraw.t.sol

```diff
@@ -0,0 +1,381 @@
+        );
+
+        // second operation
+        assertEq(
+            result.quarkOperations[1].scriptAddress,
+            paycallAddressBase,
+            "script address[1] has been wrapped with paycall address"
+        );
+        assertEq(
+            result.quarkOperations[1].scriptCalldata,
+            abi.encodeWithSelector(
+                Paycall.run.selector,
+                CodeJarHelper.getCodeAddress(type(CometWithdrawActions).creationCode),
+                abi.encodeCall(CometWithdrawActions.withdraw, (COMET, link_(8453), 5e6)),
+                1e6
+            ),
+            "calldata is Paycall.run(CometWithdrawActions.withdraw(COMET, link(8453), 5e6), 1e6);"
+        );
+        assertEq(
+            result.quarkOperations[1].expiry, BLOCK_TIMESTAMP + 7 days, "expiry is current blockTimestamp + 7 days"
+        );
+
+        // Check the actions
+        assertEq(result.actions.length, 2, "two actions");
+        // first action
+        assertEq(result.actions[0].chainId, 1, "operation is on chainid 1");
+        assertEq(result.actions[0].quarkAccount, address(0xa11ce), "0xa11ce sends the funds");
+        assertEq(result.actions[0].actionType, "BRIDGE", "action type is 'BRIDGE'");
+        assertEq(result.actions[0].paymentMethod, "PAY_CALL", "payment method is 'PAY_CALL'");
+        assertEq(result.actions[0].paymentToken, USDC_1, "payment token is USDC on mainnet");
+        assertEq(result.actions[0].paymentMaxCost, 0.1e6, "payment should have max cost of 0.1e6");
+        assertEq(
+            result.actions[0].actionContext,
+            abi.encode(
+                Actions.BridgeActionContext({
+                    amount: 1e6,
+                    price: 1e8,
+                    token: USDC_1,
+                    assetSymbol: "USDC",
+                    chainId: 1,
+                    recipient: address(0xa11ce),
+                    destinationChainId: 8453,
+                    bridgeType: Actions.BRIDGE_TYPE_CCTP
+                })
+            ),
+            "action context encoded from BridgeActionContext"
+        );
+        // second action
+        assertEq(result.actions[1].chainId, 8453, "operation is on chainid 8453");
+        assertEq(result.actions[1].quarkAccount, address(0xa11ce), "0xa11ce sends the funds");
+        assertEq(result.actions[1].actionType, "WITHDRAW", "action type is 'WITHDRAW'");
+        assertEq(result.actions[1].paymentMethod, "PAY_CALL", "payment method is 'PAY_CALL'");
+        assertEq(result.actions[1].paymentToken, USDC_8453, "payment token is USDC on Base");
+        assertEq(result.actions[1].paymentMaxCost, 1e6, "payment should have max cost of 1e6"); // ???
+        assertEq(
+            result.actions[1].actionContext,
+            abi.encode(
+                Actions.WithdrawActionContext({
+                    amount: 5e6,
+                    // assetSymbol: "USDC",
```

Comment:
not sure if we want to include `assetSymbol` and `token` fields on the `WithdrawActionContext`, but it seems like they could be useful

Reply:
I thik it's probably best if the action contexts are uniform in this way ( yae to assetSymbol )

Reply:
Agreed, let's include these. We've already gone down that path with the other action contexts

Reply:
added